.github/workflows/ci.yml

```diff
@@ -295,6 +296,23 @@ jobs:
         with:
           name-suffix: ${{ github.job }} (${{ matrix.config }})
           test-report-retention-days: ${{ env.TEST_REPORT_RETENTION_DAYS }}
+          # only upload heap dumps if running on forks, where no secrets are used
+          upload-heap-dump: >-
+            ${{ secrets.TRINO_AWS_ACCESS_KEY_ID == '' &&
+                secrets.TRINO_AWS_SECRET_ACCESS_KEY == '' &&
+                secrets.AZURE_ABFS_CONTAINER == '' &&
```

Comment:
Let's try to simplify.

Can we somehow explicitly detect the PR is not trusted?
If github doesn't make it explicit, we could check that the build is a PR and the source repo != trinodb/trino

wdyt?

(i am worried that addition of a new secret will be overlook here)

alternatively, we could have a dummy secret SECRETS_PRESENT in trinodb/trino
and condition on that here

Reply:
> alternatively, we could have a dummy secret SECRETS_PRESENT in trinodb/trino
and condition on that here

I like this idea. I didn't try to add a condition to detect forks because I'm not sure we wouldn't miss some cases. Checking for secrets seemed more direct.

Reply:
Can you create such a secret? Or would it be enough to check for just one of the existing secrets, with a comment?

Reply:
I created org-wide SECRETS_PRESENT secret.

Reply:
why is checking for PR source not enough? i.e. source repo != trinodb/trino?